src/main/java/com/uber/cadence/migration/MigrationIWorkflowService.java

```diff
@@ -140,40 +141,52 @@ private ListWorkflowExecutionsResponse appendResultsFromOldCluster(
 
     ListWorkflowExecutionsResponse fromResponse =
         callOldCluster(listWorkflowExecutionsRequest, neededPageSize, searchType);
+
+    // if old cluster is empty
+    if (fromResponse == null) {
+      return response;
+    }
+
     fromResponse.getExecutions().addAll(response.getExecutions());
     return fromResponse;
   }
 
   public boolean hasPrefix(byte[] s, byte[] prefix) {
+    if (s == null) {
+      return false;
+    }
     return s.length >= prefix.length
         && Arrays.equals(Arrays.copyOfRange(s, 0, prefix.length), prefix);
   }
 
   @Override
   public ListWorkflowExecutionsResponse ListWorkflowExecutions(
```

Comment:
could you briefly describe the List algorithm in the comment? Otherwise it's hard for others to understand.